cfy_cluster_manager/utils.py

```diff
@@ -286,9 +286,14 @@ def check_san(vm_name, vm_dict, cert_path, errors_list):
     ip_addresses = re.findall(r'\bIP Address:(\S+)\b', cert)
     dns_addresses = re.findall(r'\bDNS:(\S+)\b', cert)
     for ip in vm_dict['private_ip'], vm_dict['public_ip']:
-        if (ip in ip_addresses) and (ip in dns_addresses):
+        # adding special handle for wildcard certificate
+        if (ip in ip_addresses) and (ip in dns_addresses or
+                                     '*.{0}'.format(ip.split('.', 1)[1])
+                                     in dns_addresses):
             return
-    if hostname and hostname in dns_addresses:
+    if hostname and (hostname in dns_addresses or
+                     '*.{0}'.format(hostname.split('.', 1)[1])
```

Comment:
This'll break with an unfriendly error if hostname isn't in dns_addresses and does not contain any dots.
Rather use:
hostname.partition('.')[2]
This will always return a tuple of: (<things before divider, or everything if no divider>, <divider (if found), or ''>, <things after divider or ''>)